### PR TITLE
Stop calling top level pub

### LINKED ISSUE #88507: Use `dart pub` instead of `pub` from flutter_tools
The top-level `pub` command will be deprecated.

Part of https://github.com/dart-lang/pub/issues/2736 .

### PR DESCRIPTION
**Pull request by sigurdm**
This avoid the deprecation notice from the top level `pub` tool.

Fixes: https://github.com/flutter/flutter/issues/88507

Ideally we would shell out to `dart pub` but we want to take that step later.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Review by christopherfujino — CHANGES_REQUESTED** [review]
Thanks for the __deprecated_pub

was the license removal a mistake?

Also, what was the problem that led to the earlier version of this getting reverted?

File: examples/api/windows/flutter/generated_plugin_registrant.h [hunk 1]
```diff
@@ -1,7 +1,3 @@
-// Copyright 2014 The Flutter Authors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (sigurdm) [reply examples/api/windows/flutter/generated_plugin_registrant.h:1]: i honestly have no clue how this happened....

I reinstated the license.

**Comment by sigurdm** [comment]
> Also, what was the problem that led to the earlier version of this getting reverted?

When we made `dart pub` we did not bring over the advanced `--verbosity` options thinking that they are rarely used and can be replaced by `grep`'ping  the output from `--verbose`. 

But it turns out flutter depends on `--verbosity=warning`. Not sure about what exactly happened in the given  test case - but surely the parsing of the output of `dart pub` failed.

We might end up reconsidering this. But we would rather work on aligning `flutter pub get` so much with `dart pub get` that `flutter_tools` would not have to do its own post-processing.

**Review by christopherfujino — APPROVED** [review]
makes sense. thanks for this change, LGTM

**Pull request merged**